samples/Codelist/codelist - discriminated union.xml

```diff
@@ -101,28 +107,22 @@
 			</str:DataStructure>
 		</str:DataStructures>
 		<str:DataConstraints>
-			<str:DataConstraint urn="urn:sdmx:org.sdmx.infomodel.registry.DataConstraint=EXAMPLE:ISIC_CONSTRAINT(1.0)" isExternalReference="false" agencyID="EXAMPLE" id="ISIC_CONSTRAINT" role="Allowed" version="1.0">
+			<str:DataConstraint urn="urn:sdmx:org.sdmx.infomodel.registry.DataConstraint=EXAMPLE:ISIC_CONSTRAINT(1.0)" isExternalReference="false" agencyID="EXAMPLE" id="ISIC_CONSTRAINT" version="1.0">
 				<com:Name xml:lang="en">ISIC Constraint</com:Name>
 				<com:Description xml:lang="en">ISIC Constraint</com:Description>
-				<str:ConstraintAttachment>
-				<str:Dataflow>urn:sdmx:org.sdmx.infomodel.datastructure.Dataflow=EXAMPLE:LABOUR_ISIC(1.0)</str:Dataflow>
-				</str:ConstraintAttachment>
 				<str:CubeRegion include="true">
-					<str:KeyValue id="ACTIVITY">
+					<str:Component id="ACTIVITY">

```

Comment:
Finally, `Keyvalue` was not replaced by `Component` thus this should be reverted.

Reply:
https://github.com/sdmx-twg/sdmx-ml/blob/3_1_0_versionchange/schemas/SDMXStructureConstraint.xsd

Line 302 - element name=Component 

This file as it is currently is valid wrt the schema as it is in this branch

Reply:
The viz tool I used was buggy and misleading me. I see now that CubeRegion only allows for Components. The example is thus ok. I wonder if it will not be confusing for users that Components in CubeRegions can hold any component while those in Keys can only hold attributes.

Reply:
Thanks - I was wondering the same thing just now as well - do you have a suggestion for an alternative name (I can't think of one)

Reply:
A solution might be to rename Component to Attribute in the Keys (https://github.com/sdmx-twg/sdmx-ml/blob/18cfd3e38b0c31476d4e1657ea3520c36cf9195c/schemas/SDMXStructureConstraint.xsd#L256)

Reply:
I can make this change, my only note on this would be that it also includes Measures - and I don't think there is a collective term for both Attribute and Measures - but perhaps Attribute is less confusing

Reply:
@agent96 You are right. Then maybe instead of `Attribute` it could be `NonKeyComponents` or `NonDimensionComponents`, but this doesn't look very appealing either... 🤔